components/benefit_sponsors/app/models/effective/datatables/benefit_sponsors_employer_datatable.rb

```diff
@@ -270,6 +244,95 @@ def nested_filter_definition
       def authorized?(current_user, _controller, _action, _resource)
         current_user.present? && current_user.has_hbx_staff_role?
       end
+
+      def collection
+        return @collection if defined? @collection
+
+        benefit_sponsorships ||= BenefitSponsors::BenefitSponsorships::BenefitSponsorship.unscoped
+        benefit_sponsorships = filter_by_employers(benefit_sponsorships)
+        benefit_sponsorships = filter_by_enrolling(benefit_sponsorships)
+        benefit_sponsorships = filter_by_enrolled(benefit_sponsorships)
+        benefit_sponsorships = filter_by_employer_attestations(benefit_sponsorships)
+        benefit_sponsorships = filter_by_upcoming_dates(benefit_sponsorships)
+        benefit_sponsorships = filter_by_attestations(benefit_sponsorships)
+
+        @collection = benefit_sponsorships
+      end
+
+      private
+
+      def filter_by_employers(benefit_sponsorships)
+        return benefit_sponsorships unless attributes[:employers].present? && !['all'].include?(attributes[:employers])
+
+        call_safe_method(benefit_sponsorships, attributes[:employers]) if ALLOWED_METHODS.include?(attributes[:employers])
+      end
+
+      def filter_by_enrolling(benefit_sponsorships)
+        return benefit_sponsorships unless attributes[:enrolling].present?
+
+        if attributes[:enrolling_initial].present? || attributes[:enrolling_renewing].present?
+          benefit_sponsorships = call_safe_method(benefit_sponsorships, attributes[:enrolling_initial]) if attributes[:enrolling_initial].present? && attributes[:enrolling_initial] != 'all'
+          benefit_sponsorships = call_safe_method(benefit_sponsorships, attributes[:enrolling_renewing]) if attributes[:enrolling_renewing].present? && attributes[:enrolling_renewing] != 'all'
+          call_safe_method(benefit_sponsorships, attributes[:enrolling]) if attributes[:enrolling_initial] == 'all' || attributes[:enrolling_renewing] == 'all'
+        else
+          call_safe_method(benefit_sponsorships, attributes[:enrolling])
+        end
+      end
+
+      def filter_by_enrolled(benefit_sponsorships)
+        attributes[:enrolled].present? ? call_safe_method(benefit_sponsorships, attributes[:enrolled]) : benefit_sponsorships
+      end
+
+      def filter_by_employer_attestations(benefit_sponsorships)
+        return benefit_sponsorships unless attributes[:employer_attestations].present?
+
+        benefit_sponsorships = call_safe_method(benefit_sponsorships, attributes[:employer_attestations])
+        case attributes[:attestation_status]
```

Comment:
Seems like it's new logic. Why do we add it here?

Reply:
Thank you for your observation! The logic to call call_safe_method for attributes[:employer_attestations] is indeed a new addition. This was added to ensure that employer attestations are handled consistently and securely by validating them against the allowed methods list. I am open to further discussions or adjustments to align with our requirements. Your feedback is much appreciated!